src/dotnet/MsbuildSolution.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools
+{
+    internal class MsbuildSolution
```

Comment:
So there is another class (that already existed before this PR) called MsbuildProject. I called this one MsbuildSolution to be consistent.